/*
 * You may redistribute this program and/or modify it under the terms of
 * the GNU General Public License as published by the Free Software Foundation,
 * either version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

syntax = "proto2";
package p2pnet.protocol;

message OverlayMessage {
	message Header {
		required bytes src_th = 1;
		optional bytes dest_th = 2;

		//enum MessagePriority {
		//	REALTIME = 0;	// This is like UDP.
		//	RELIABLE = 1;	// This is like SCTP (RFC 4960), but only with delivery control.
		//}

		//optional MessagePriority prio = 3 [default = RELIABLE];
		//optional uint32 seq_num = 4;
		//repeated uint32 ack_num = 5 [packed = true];
	}
	required Header header = 1;

	message Payload {
		required uint32 type = 1;
		required bytes content = 2;
	}
	message MultiPayload {
		repeated Payload payload = 1;
	}
	message Body {
		optional MultiPayload open_multipayload = 1;
		optional bytes encrypted_multipayload = 2;
	}

	required Body body = 2;
}